crates/storage-pg/src/compat/session.rs

```diff
@@ -341,6 +341,64 @@ impl<'c> CompatSessionRepository for PgCompatSessionRepository<'c> {
         Ok(compat_session)
     }
 
+    #[tracing::instrument(
+        name = "db.compat_session.finish_bulk",
+        skip_all,
+        fields(db.statement),
+        err,
+    )]
+    async fn finish_bulk(
+        &mut self,
+        clock: &dyn Clock,
+        filter: CompatSessionFilter<'_>,
+    ) -> Result<usize, Self::Error> {
+        let finished_at = clock.now();
+        let (sql, arguments) = Query::update()
+            .table(CompatSessions::Table)
+            .value(CompatSessions::FinishedAt, finished_at)
+            .and_where_option(filter.user().map(|user| {
```

Comment:
should the `CompatSessionFilter` be responsible for constructing its own `WHERE` clauses? I am a bit concerned we'd forget to update this code here if we added more to the filter

Reply:
I did think of this, but it was annoying in particular for the compatibility sessions, as when we query, we `JOIN` on the `compat_sso_logins` and filter on that, which we can't do for updating the `finished_at` field.

I'll have a shot though in another PR